### PR TITLE
hash.dev: switch to next-seo, add default OG image

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

The motivation for this PR is to add a default OG image to all pages, as currently only the individual blog pages have them.

It introduces `next-seo` as the easiest way of managing metadata defaults + overwrites.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203301853897838/1203358502199028/f) _(internal)_

## ❓ How to test this?

1.  View the index, blog index, and individual blog pages – check the elements in `<head>`
1.  Once the preview deployment is available, paste a link to pages in Slack (or wherever), check the preview
1.  Paste a link in a draft tweet

### CODE REVIEW

File: sites/hashdev/src/pages/_app.page.tsx [hunk 1]
```diff
@@ -5,12 +5,30 @@ import "../../styles/prism.css";
...
+const defaultSeoProps: DefaultSeoProps = {
+  title: "HASH.dev – HASH for Developers",
+  description: SITE_DESCRIPTION,
+  twitter: {
+    cardType: "summary_large_image",
+    site: "@hashintel",
+  },
+  openGraph: {
+    // title: siteTitle,
+    // description: siteDescription,
```
Comment (yusufkinatas) [inline sites/hashdev/src/pages/_app.page.tsx:26]: do we need these lines?

**Review by yusufkinatas — APPROVED** [review]

File: sites/hashdev/src/pages/_app.page.tsx (hunk 1 above)
Comment (CiaranMn) [reply sites/hashdev/src/pages/_app.page.tsx]: Oops, thank you! Removed

File: sites/hashdev/src/pages/_app.page.tsx [hunk 2]
```diff
@@ -5,12 +5,28 @@ import "../../styles/prism.css";
...
 import { MuiProvider } from "../theme/MuiProvider";
 import { NextPageWithLayout } from "../util/nextTypes";
+import { FRONTEND_URL, SITE_DESCRIPTION } from "../config";
+
+const defaultSeoProps: DefaultSeoProps = {
+  title: "HASH.dev – HASH for Developers",
+  description: SITE_DESCRIPTION,
+  twitter: {
+    cardType: "summary_large_image",
+    site: "@hashintel",
```
Comment (nonparibus) [inline sites/hashdev/src/pages/_app.page.tsx:22]: We can use `@hashdevs` here

Comment (CiaranMn) [reply sites/hashdev/src/pages/_app.page.tsx]: Thanks - updated in https://github.com/hashintel/hash/pull/1487/commits/9e1e9217287ea9bcde0d0d193f03da90e177845a

**Pull request merged**